vault/datadog_checks/vault/vault.py

```diff
@@ -240,17 +237,25 @@ def parse_config(self):
             instance['ssl_ca_cert'] = instance.pop('tls_ca_cert', None)
 
             self._scraper_config = self.create_scraper_configuration(instance)
-            self.set_client_token(self._client_token)
 
             # Global static tags
             self._scraper_config['custom_tags'] = self._tags
 
+            # https://www.vaultproject.io/api/overview#the-x-vault-request-header
+            self._set_header(self.http_handlers[instance['prometheus_url']], 'X-Vault-Request', 'true')
+
+            if self._client_token_path:
+                self.renew_client_token()
+            else:
+                self.set_client_token(self._client_token)
```

Comment:
Why changes are needed in `vault` ?

Reply:
It had to do with where `create_scraper_configuration` was called

Reply:
I mean, why 

```
Actually, I’m not sure, why 

```
if self._client_token_path:
                self.renew_client_token()
``` 

has been moved inside the `if condition` old line 232.

Reply:
`renew_client_token` calls `set_client_token`, `set_client_token` updates `self.http_handlers[self._scraper_config['prometheus_url']`, `prometheus_url` wasn't set until next block